### ISSUE TITLE
Datasets conflicts with fsspec 2024.9

### ISSUE THREAD

**Issue by cw-igormorgado** [huggingface/datasets#7190]
### Describe the bug

Installing both in latest versions are not possible

`pip install "datasets==3.0.1" "fsspec==2024.9.0"`

But using older version of datasets is ok

`pip install "datasets==1.24.4" "fsspec==2024.9.0"`

### Steps to reproduce the bug

`pip install "datasets==3.0.1" "fsspec==2024.9.0"`

### Expected behavior

install both versions.

### Environment info

debian 11.
python 3.10.15

**Comment by geekodour** [comment]
Yes, I need to use the latest version of fsspec and datasets for my usecase. 
https://github.com/fsspec/s3fs/pull/888#issuecomment-2404204606
https://github.com/apache/arrow/issues/34363#issuecomment-2403553473
